build/authors.py

```diff
@@ -2,143 +2,21 @@
 
-    return formatted_info
 
-def get_affiliations(author_df):
+def get_author_info(path):
     """
-    Takes an author data frame in which authors have already been ordered and
-    creates affiliation mappings.  Each unique affiliation is assigned an
-    index based on the order in which it appears in the ordered author list.
-    Complete affiliation strings are used to determine affiliation uniqueness
-    even if on string contains multiple departments or institutions.
+    Load the author table and return the list of authors.
     """
-    # Each author is a dict with a name, affiliation index, and symbol that
-    # indicates corresponding or other special notes
-    authors = list()
-    affiliations = list()
-    corresponding = list()
-    affiliation_map = dict()
-    affiliation_counter = 1
-
-    for (_, row) in author_df.iterrows():
-        name = row['full_name']
-        affiliation = row['affiliation']
-
-        # Lookup the affiliation index or create it
-        # Only list each affiliation once
-        if not affiliation in affiliation_map:
-            affiliation_map[affiliation] = affiliation_counter
-            affiliation_info = OrderedDict()
-            affiliation_info['index'] = affiliation_counter
-            affiliation_info['institution'] = affiliation
-            affiliations.append(affiliation_info)
-            affiliation_counter += 1
-        index = affiliation_map[affiliation]
-
-        # Add the affiliation superscript
-        # Assumes a single affiliation per author (which could contain multiple
-        # institutions) but could be generalized
-        author = OrderedDict()
-        author['name'] = name
-        author['affiliation'] = index
-        if pandas.notnull(row['corresponding']) and row['corresponding'].lower() == 'yes':
-            author['symbol'] = ',†'
-            corresponding.append(f'{row["email"]} ({row["initials"]})')
-        else:
-            author['symbol'] = ''
-        authors.append(author)
-
-    # Add special annotation for the first listed author
-    first = authors[0]
-    first['symbol'] = first['symbol'] + ',*'
-
-    affiliation_info = OrderedDict()
-    affiliation_info['authors'] = authors
-    affiliation_info['affiliations'] = affiliations
-    affiliation_info['corresponding'] = format_list(corresponding)
-    return affiliation_info
-
-def get_author_info(author_file):
-    """
-    Load the author table and return a dict with author information such as
-    the number of authors.
-    """
-    author_df = pandas.read_table(author_file)
-
-    # Add author initials
-    author_df['initials'] = author_df['full_name'].apply(initials)
-
-    # Avoid initial collisions, will need a more general way to refer to
-    # authors if they occur
-    assert len(author_df['initials'].unique()) == len(author_df), \
-        'All authors must have unique initials'
-    assert np.all(author_df['approve'].str.match('Yes', case=False)), \
```

Comment:
I'm mostly trying to make some notes as we proceed.  We may not want to actually write all the docs until the build system stabilizes.